### PR TITLE
Implement and use ch_chirp_release_recv_handler

### PR DESCRIPTION
**Pull request by ganwell**
* Added CH_MSG_IS_HANDLER flag to check consistency, preventing users
  from releasing non handler messages
* Use check if reader->handler is NULL instead of indirect check if we
  at a new message by checking bytes_handled
* Release ACK messages directly

### CODE REVIEW

File: src/reader.c [hunk 1]
```diff
@@ -558,7 +557,36 @@ ch_rd_read(ch_connection_t* conn, void* buffer, size_t read)
...
+    ch_connection_t* conn = msg->_conn;
+    ch_chirp_t* chirp = conn->chirp;
+    A(chirp->_init == CH_CHIRP_MAGIC, "Not a ch_chirp_t*");
+    ch_chirp_int_t* ichirp = chirp->_;
+    if(msg->_flags & CH_MSG_FREE_DATA)
+        ch_free(msg->data);
+    if(msg->_flags & CH_MSG_FREE_HEADER)
+        ch_free(msg->header);
+    ch_bf_release(&ichirp->pool, msg->_handler);
+    if(conn->reader.last) {
```
Comment (sliverc) [inline src/reader.c]: More a question than a comment:
What can actually make the last flag to be true? I am just a bit confused about the name.

**Comment by sliverc** [comment]
@ganwell CI is failing... otherwise looks good just added a comment. I will be on vacation till October so I guess @winged can potentially look at your PRs.

File: src/reader.c [hunk 2]
```diff
@@ -450,7 +449,7 @@ ch_rd_read(ch_connection_t* conn, void* buffer, size_t read)
                 if(reader->handler == NULL) {
                     reader->handler = ch_bf_acquire(
                         &ichirp->pool,
-                        &reader->last
+                        &reader->last_handler
```
Comment (ganwell) [inline src/reader.c]: @sliverc this is where last_handler is set

**Comment by ganwell** [comment]
@sliverc I added two commit

1. Ignore sphinx highlighting failing on valid C code
2. Renamed last to last_handler

**Pull request merged**